airlock/management/commands/run_file_uploader.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import os
+import time
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from opentelemetry import trace
+
+import old_api
+from airlock.business_logic import bll
+from airlock.enums import RequestStatus
+from airlock.users import User
+from services.tracing import instrument
+
+
+logger = logging.getLogger(__name__)
+
+# We need a user with the output-checker role to access some
+# bll methods
+system_user = User("system", output_checker=True)
+
+
+class Command(BaseCommand):
+    """
+    Poll database for approved requests that still have files pending upload
+    """
+
+    def add_arguments(self, parser):
+        # In production, we want this check to run forever. Using a
+        # function means that we can test it on a finite number of loops.
+        parser.add_argument("--run-fn", default=lambda: True)
+
+    def handle(self, *args, **options):
+        run_fn = options["run_fn"]
+
+        logger.debug("Watching for tasks")
+
+        while run_fn():  # pragma: no branch
```

Comment:
I think this may need a top level exception handler, to log and re-launch.

We don't know where an exception might come from. e.g. maybe bll.get_approved_requests() throws a sqlite locked error, which is a temporary error.

AIUI, in that case the management command will exit with an error.  And we would need to restart the entire container to restart it again.

I think this management command should do its best to stay running in the face of errors. Basically, it kind of needs to act as its own process supervisor (like gunicorn supervises the web workers, and restarts one if it errors).  Or else we may need to manually log in to L4 and restart the container if something goes wrong?

Reply:
Could we have a bash script wrapper that does this and then use that as the entrypoint? Rather than bundling the complexity into the script itself. The fact that it needs to worry about this is an artefact of not having a proper supervisor for it at the moment. And it might be nicer to stop that detail leaking into the code itself.

Reply:
Yes, we can. And if it restarts the process from scratch, we gain better protection from memory leaks or SIGSEVs too.

I think that the built in init (`tini`) can do this for us, I will try experiment

Reply:
Solved by using run-one-constantly in the entrypoint